src/main/java/org/qiyana/model/LineItem.java

```diff
@@ -0,0 +1,59 @@
+package org.qiyana.model;
+
+import javax.persistence.*;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+
+import java.util.Currency;
+
+@Entity
+@NoArgsConstructor
+@Getter
+public class LineItem {
+  @Id
+  @GeneratedValue(strategy = GenerationType.IDENTITY)
+  private Long id;
+
+  private Integer quantity;
+  private double totalItemPrice;
+  private Currency currency;
+
+  @ManyToOne
+  @JoinColumn(name = "order_id")
+  private Order order;
+
+  @OneToOne
+  @JoinColumn(name = "product_id", referencedColumnName = "id")
+  private Product product;
+
+  public LineItem(
+      Long id, Integer quantity,String currencyCode, Product product, Order order) {
+    this.id = id;
```

Comment:
`id` and any other `GeneratedValue` should never be a constructor parameter. You will pretty much always pass `null` here.